FakeNFT/Scenes/NFTCollectionDetail/NFTCollectionDetailPresenter.swift

```diff
@@ -5,19 +5,142 @@
 //  Created by Nikolay on 20.02.2025
 //
 
+import Foundation
+
 protocol NFTCollectionDetailPresenterProtocol: AnyObject {
+    func loadCurrentNFTCollection()
+    func showAuthorPage()
+    func nftCount() -> Int
+    func nftBusinessObject(index: IndexPath) -> NftBO?
+    func loadProfile()
 }
 
 final class NFTCollectionDetailPresenter {
     weak var view: NFTCollectionDetailViewProtocol?
     var router: NFTCollectionDetailRouterProtocol
     var interactor: NFTCollectionDetailInteractorProtocol
+    var input: NftCollectionDetailInput
+    
+    private var currentNftCollection: NftCollection?
+    private var users: [User]?
+    private var authorWebsiteURL: URL?
+    private var nfts: [Nft]? = []
+    private var profile: Profile?
+    private var order: Order?
 
-    init(interactor: NFTCollectionDetailInteractorProtocol, router: NFTCollectionDetailRouterProtocol) {
+    init(interactor: NFTCollectionDetailInteractorProtocol, router: NFTCollectionDetailRouterProtocol, input: NftCollectionDetailInput) {
         self.interactor = interactor
         self.router = router
+        self.input = input
     }
 }
 
 extension NFTCollectionDetailPresenter: NFTCollectionDetailPresenterProtocol {
+    
+    func loadCurrentNFTCollection() {
+        interactor.loadNftCollection(id: input.id) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let nftCollection):
+                self.loadUsers()
+                self.currentNftCollection = nftCollection
+                let url = URL(string: nftCollection.cover.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? "")
+                self.view?.updateNftCollectionInformation(name: nftCollection.name.capitalized,
+                                                          imageURL: url,
+                                                          description: nftCollection.description.capitalized, authorName: nftCollection.author)
```

Comment:
Так поудобнее читать, как будто 
```suggestion
                self.view?.updateNftCollectionInformation(
                    name: nftCollection.name.capitalized,
                    imageURL: url,
                    description: nftCollection.description.capitalized, 
                    authorName: nftCollection.author
                )
```